### PR TITLE
Update events.lua to fix robot follower count bug

### PR DESCRIPTION
**Pull request by Jayefuu**
Checks that there is an artillery wagon near upon robot follower timeout before creating a cluster grenade projectile.

### CODE REVIEW

File: map_gen/maps/crash_site/events.lua [hunk 1]
```diff
@@ -487,9 +487,12 @@ Event.add(
         local entity = event.robot
         local position = entity.position
         if entity.force.name == 'enemy' then
-            entity.surface.create_entity{name = "cluster-grenade", position=position, target=position, speed=1}
+            -- only create a grenade entity if there is an artillery wagon near, otherwise the maximum robot follower count will spoil your fun
```
Comment (grilledham) [inline map_gen/maps/crash_site/events.lua]: Can you get this from `event.owner`? <https://lua-api.factorio.com/latest/events.html#on_combat_robot_expired>
Failing that count_entities_filtered <https://lua-api.factorio.com/latest/LuaSurface.html#LuaSurface.count_entities_filtered> might be a better choice.

File: map_gen/maps/crash_site/events.lua [hunk 2]
```diff
@@ -483,13 +483,11 @@ Event.add(
...
     function(event)
-
         local entity = event.robot
         local position = entity.position
         if entity.force.name == 'enemy' then
-            -- only create a grenade entity if there is an artillery wagon near, otherwise the maximum robot follower count will spoil your fun
-            local wagon_entities = entity.surface.find_entities_filtered {name="artillery-wagon", position=position, radius=15} 
-            if #wagon_entities > 0 then
+            -- only create a grenade entity if an artillery wagon (the event.owner) killed the target that spawned the combabt robot
+            if event.owner.name == "artillery-wagon"
```
Comment (grilledham) [inline map_gen/maps/crash_site/events.lua]: owner could be nil or invalid so we will have to check that.

File: map_gen/maps/crash_site/events.lua [hunk 3]
```diff
@@ -483,13 +483,17 @@ Event.add(
...
         local entity = event.robot
         local position = entity.position
+        local owner = event.owner
+        if owner == nil or not owner.valid then
+            return
+        end
         if entity.force.name == 'enemy' then
-            entity.surface.create_entity{name = "cluster-grenade", position=position, target=position, speed=1}
+            -- only create a grenade entity if an artillery wagon (the event owner) killed the target that spawned the combabt robot
+            if owner.name == "artillery-wagon" then
```
Comment (grilledham) [inline map_gen/maps/crash_site/events.lua]: You could merge this with the previous `if`, but that is just a matter of style.

**Review by grilledham — APPROVED** [review]

File: map_gen/maps/crash_site/events.lua (hunk 3 above)
Comment (Jayefuu) [reply map_gen/maps/crash_site/events.lua]: You're right it's neater, it was left over from some tests. I've changed it as suggested.

**Pull request merged**